pkg/test/framework/components/apps/kube.go

```diff
@@ -593,6 +603,12 @@ func (a *kubeApp) Call(e AppEndpoint, opts AppCallOptions) ([]*echo.ParsedRespon
 	// Forward a request from 'this' service to the destination service.
 	dstURL := dst.makeURL(opts)
 	dstServiceName := dst.owner.Name()
+
+	// If host header is set, override the destination with it
+	if opts.Headers.Get("Host") != "" {
+		dstServiceName = opts.Headers.Get("Host")
+	}
+
 	resp, err := a.client.ForwardEcho(&proto.ForwardEchoRequest{
 		Url:   dstURL.String(),
```

Comment:
Does this need to be changed as well?

Reply:
Yes, this basically just allows for the overriding of the host which is required to fudge the request to take into account locality via a service entry.

Reply:
No, Envoy throws away the resolved IP.